pkg/nodesconfig/nodesconfig.go

```diff
@@ -0,0 +1,195 @@
+
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient is empty")
+
+		return nil, fmt.Errorf("nodesConfig Config 'apiClient' cannot be empty")
+	}
+
+	if nodesConfigObjName == "" {
+		glog.V(100).Infof("The name of the nodesConfig is empty")
+
+		return nil, fmt.Errorf("nodesConfig 'nodesConfigObjName' cannot be empty")
+	}
+
+	builder := Builder{
+		apiClient: apiClient.Client,
+		Definition: &configV1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: nodesConfigObjName,
+			},
+		},
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("nodesConfig object %s doesn't exist", nodesConfigObjName)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Get fetches existing nodesConfig from cluster.
+func (builder *Builder) Get() (*configV1.Node, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting existing nodesConfig with name %s from cluster", builder.Definition.Name)
+
+	nodesConfig := &configV1.Node{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, nodesConfig)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to get nodesConfig object %s from cluster due to: %w",
+			builder.Definition.Name, err)
+
+		return nil, err
+	}
+
+	return nodesConfig, nil
+}
+
+// Exists checks whether the given nodesConfig exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if nodesConfig %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Update renovates the nodesConfig in the cluster and stores the created object in struct.
+func (builder *Builder) Update() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating the nodesConfig %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("nodesConfig object %s doesn't exist", builder.Definition.Name)
+	}
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+	if err == nil {
+		builder.Object = builder.Definition
+	}
+
+	return builder, err
+}
+
+// GetCGroupMode fetches nodesConfig cgroupMode.
+func (builder *Builder) GetCGroupMode() (configV1.CgroupMode, error) {
+	if valid, err := builder.validate(); !valid {
+		return "", err
+	}
+
+	glog.V(100).Infof("Getting nodesConfig ManagementState configuration")
+
+	if !builder.Exists() {
+		return "", fmt.Errorf("imageRegistry object doesn't exist")
+	}
+
+	return builder.Object.Spec.CgroupMode, nil
+}
+
+// WithCGroupMode sets the nodesConfig operator's cgroupMode.
+func (builder *Builder) WithCGroupMode(expectedCGroupMode configV1.CgroupMode) *Builder {
+	glog.V(100).Infof("Setting nodesConfig %s with ManagementState: %v",
+		builder.Definition.Name, expectedCGroupMode)
```

Comment:
Please add validate block before this

Reply:
we have it just after the glog. I'll switch them...

Reply:
Yeah but the validate method protects us from nil pointer exceptions